code/model/modelanimation.cpp

```diff
@@ -213,70 +329,219 @@ namespace animation {
+				angle.p = fl_radians(anglesDeg.xyz.x);
+				angle.h = fl_radians(anglesDeg.xyz.y);
+				angle.b = fl_radians(anglesDeg.xyz.z);
+
+				isRelative = true;
+			}
+
+			if (optional_string("+velocity:"))
+				skip_token();
+
+			if (optional_string("+acceleration:"))
+				skip_token();
+
+			if (optional_string("+time:"))
+				skip_token();
+
+			std::shared_ptr<ModelAnimation> anim = ModelAnimation::createAnimation(true);
+
+			char namelower[MAX_NAME_LEN];
+			strncpy(namelower, sp->subobj_name, MAX_NAME_LEN);
+			strlwr(namelower);
+			//since sp->type is not set without reading the pof, we need to infer it by subsystem name (which works, since the same name is used to match the submodels name, which is used to match the type in pof parsing)
+			//sadly, we also need to check for engine and radar, since these take precedent (as in, an engineturret is an engine before a turret type)
+			if (!strstr(namelower, "engine") && !strstr(namelower, "radar") && strstr(namelower, "turret")) {
+				auto rotBase = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.h));
+				auto subsysBase = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, false, sip->name, std::move(rotBase)));
+				anim->addSubmodelAnimation(std::move(subsysBase));
+
+				auto rotBarrel = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.p));
+				auto subsysBarrel = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, true, sip->name, std::move(rotBarrel)));
+				anim->addSubmodelAnimation(std::move(subsysBarrel));
+			}
+			else {
+				auto rot = std::shared_ptr<ModelAnimationSegmentSetPHB>(new ModelAnimationSegmentSetPHB(angle, isRelative));
+				auto subsys = std::shared_ptr<ModelAnimationSubmodel>(new ModelAnimationSubmodel(sp->subobj_name, std::move(rot)));
+				anim->addSubmodelAnimation(std::move(subsys));
+			}
 
-			auto rotBarrel = std::shared_ptr<ModelAnimationSegmentSetAngle>(new ModelAnimationSegmentSetAngle(angle.p));
-			auto subsysBarrel = std::shared_ptr<ModelAnimationSubmodelTurret>(new ModelAnimationSubmodelTurret(sp->subobj_name, true, sip->name, std::move(rotBarrel)));
-			anim->addSubmodelAnimation(std::move(subsysBarrel));
+			sip->animations.emplace(anim, anim_name_from_subsys(sp), animation::ModelAnimationTriggerType::Initial);
 		}
 		else {
-			auto rot = std::shared_ptr<ModelAnimationSegmentSetPHB>(new ModelAnimationSegmentSetPHB(angle, isRelative));
-			auto subsys = std::shared_ptr<ModelAnimationSubmodel>(new ModelAnimationSubmodel(sp->subobj_name, std::move(rot)));
-			anim->addSubmodelAnimation(std::move(subsys));
-		}
+			std::shared_ptr<ModelAnimation> anim = ModelAnimation::createAnimation();
+
+			if (type == ModelAnimationTriggerType::TurretFired) {
+				//Turret fireds won't get reset by code, so make them auto-resetting
+				anim->m_flags += Animation_Flags::Auto_Reverse;
+			}
```

Comment:
Right, I intend to add another flag called "reset at stop", which is essentially puts the animation back to it's untriggered state once it completes, and is useful for behaviour where the animation ends where it starts (since you probably don't want to reverse from there).
Then we can throw a warning (or even error here), if a turret fired animation has neither flag.